server/controllers/auth.js

```diff
@@ -0,0 +1,73 @@
+const bcrypt = require('bcrypt');
+const jwt = require('jsonwebtoken');
+
+const {
+  getArtistByEmail,
+  addArtist,
+  addCustomer,
+  checkCustomerEmail,
+} = require('../database/queries');
+
+const registerController = async (req, res, next) => {
+  const {
+    body: { firstName, lastName, password, email, role },
+  } = req;
+  try {
+    let existingUser;
+    switch (role) {
+      case 'artist':
+        existingUser = await getArtistByEmail(email);
+        break;
+
+      case 'customer':
+        existingUser = await checkCustomerEmail(email);
+        break;
+      default:
+        throw new Error('Choose your role');
+    }
+    if (existingUser.rows.length !== 0)
+      return res
+        .status(400)
+        .json({ statusCode: 400, message: 'email exist !!' });
+
+    let newPassword;
+    try {
+      const hash = await bcrypt.hash(password, 10);
+      newPassword = hash;
+    } catch (error) {
+      next(error);
+    }
+    const newUser = {
+      firstName,
+      lastName,
+      email,
+      password: newPassword,
+    };
+    const addUser = async (query, user, roleUser) => {
+      try {
+        const { rows } = await query(user);
+        const { id, name } = rows[0];
+        const token = jwt.sign({ id, name }, process.env.SECRET_KEY);
```

Comment:
That's not the way I send cookies, just send Id and role as a token and name it token just, because we want the role at cookies encrypted so no one can edit it.